pkg/compliance/agent/node_labels.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package agent
+
+import (
+	"math"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/hostinfo"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/cenkalti/backoff"
+)
+
+const (
+	nodeLabelsCheckInitialInterval = time.Second
+	nodeLabelsCheckMaxInterval     = 10 * time.Second
+	nodeLabelsCheckMultiplier      = 2.0
+	nodeLabelsCheckMaxElapsedTime  = 1 * time.Minute
+)
+
+// WaitGetNodeLabels waits for node labels to become available using a backoff retrier
+func WaitGetNodeLabels() (map[string]string, error) {
+	fetcher := &labelsFetcher{}
+	exp := &backoff.ExponentialBackOff{
+		InitialInterval:     nodeLabelsCheckInitialInterval,
+		RandomizationFactor: 0,
+		Multiplier:          nodeLabelsCheckMultiplier,
+		MaxInterval:         nodeLabelsCheckMaxInterval,
+		MaxElapsedTime:      nodeLabelsCheckMaxElapsedTime,
+		Clock:               backoff.SystemClock,
+	}
+	exp.Reset()
+	err := backoff.RetryNotify(fetcher.fetch, exp, notifyFetchNodeLabels())
```

Comment:
I think we have `github.com/DataDog/datadog-agent/pkg/util/retry` to do that already

Reply:
Yes, this is also already used in WPA controller. It's a bit easier to work with.